src/rt/cover.c

```diff
@@ -45,6 +45,13 @@ enum std_ulogic {
    _DC = 0x8
 };
 
+#define INCS_I32(i32_ptr)                          \
+      do {                                         \
+         int32_t inc = *i32_ptr + 1;               \
+         if (likely(inc > *i32_ptr))               \
+            *i32_ptr = inc;                        \
```

Comment:
This doesn't work with optimisation turned on as signed types have undefined behaviour on overflow unlike unsigned types which wrap. GCC and and Clang remove the saturation check completely: https://godbolt.org/z/hY7xjaK5T

```suggestion
         if (unlikely(__builtin_add_overflow(*i32_ptr, 1, i32_ptr))) \
            *i32_ptr = INT32_MAX;                                    \
```

Reply:
Thanks. I don't have that much experience with the nuances of the C standard, this is very much appreciated.